ui/hooks/subscription/useSubscription.ts

```diff
@@ -282,3 +297,130 @@ export const useSubscriptionEligibility = (product: ProductType) => {
+  useTestClock?: boolean;
+}) => {
+  const dispatch = useDispatch<MetaMaskReduxDispatch>();
+  const { execute: executeSubscriptionCryptoApprovalTransaction } =
+    useSubscriptionCryptoApprovalTransaction(selectedToken);
+  const { subscriptions, lastSubscription } = useUserSubscriptions();
+  const { captureShieldSubscriptionRequestEvent } = useSubscriptionMetrics();
+  const modalType: ModalType = useSelector(getModalTypeForShieldEntryModal);
+
+  const latestSubscriptionStatus =
+    getLatestSubscriptionStatus(subscriptions, lastSubscription) || 'none';
+
+  const [handleSubscription, subscriptionResult] =
+    useAsyncCallback(async () => {
+      // save the last used subscription payment method and plan to Redux store
+      await dispatch(
+        setLastUsedSubscriptionPaymentDetails(PRODUCT_TYPES.SHIELD, {
+          type: selectedPaymentMethod,
+          paymentTokenAddress: selectedToken?.address as Hex,
+          paymentTokenSymbol: selectedToken?.symbol,
+          plan: selectedPlan,
+          useTestClock,
+        }),
+      );
+
+      const subscriptionRequestTrackingParams: Omit<
+        CaptureShieldSubscriptionRequestParams,
+        'requestStatus'
+      > = {
+        subscriptionState: latestSubscriptionStatus,
+        defaultPaymentType: defaultOptions.defaultPaymentType,
+        defaultPaymentCurrency: defaultOptions.defaultPaymentCurrency,
+        defaultBillingInterval: defaultOptions.defaultBillingInterval,
+        defaultPaymentChain: defaultOptions.defaultPaymentChain,
+        paymentType: selectedPaymentMethod,
+        paymentCurrency: 'USD',
+        isTrialSubscription: !isTrialed,
+        billingInterval: selectedPlan,
+        source: EntryModalSourceEnum.Settings,
+        type: modalType,
+      };
+
+      if (selectedPaymentMethod === PAYMENT_TYPES.byCard) {
+        try {
+          // capture the event when the Shield subscription request is started
+          captureShieldSubscriptionRequestEvent({
+            ...subscriptionRequestTrackingParams,
+            paymentCurrency: 'USD',
+            requestStatus: 'started',
+          });
+          await dispatch(
+            startSubscriptionWithCard({
+              products: [PRODUCT_TYPES.SHIELD],
+              isTrialRequested: !isTrialed,
+              recurringInterval: selectedPlan,
+              useTestClock,
+            }),
+          );
+          // capture the event when the Shield subscription request is completed
+          captureShieldSubscriptionRequestEvent({
+            ...subscriptionRequestTrackingParams,
+            paymentCurrency: 'USD',
+            requestStatus: 'completed',
+          });
+        } catch (error) {
+          const errorMessage =
+            error instanceof Error ? error.message : 'Unknown error';
+          // capture the event when the Shield subscription request fails
+          captureShieldSubscriptionRequestEvent({
+            ...subscriptionRequestTrackingParams,
```

Comment:
cannot do this. should be moved to where controller method startSubscription is called and completed